<launch>
  <machine name="local" address="localhost" default="true" />

  <param name="use_sim_time" value="true"/>
  <arg name="dataset" default="" />
  <arg name="rate" default="1.0" />

  <node name="tinySLAM" pkg="slam_fmwk" type="tiny_slam"
        cwd="node" required="true" output="screen">
    <remap from="laser_scan" to="/base_scan" />

    <!-- Modules setup -->
    <param name="slam/cell/type" value="avg" />

    <!-- Occupancy estimator setup -->
    <param name="slam/occupancy_estimator/type" value="const" />
    <param name="slam/occupancy_estimator/base_occupied/prob" value="0.95" />
    <param name="slam/occupancy_estimator/base_empty/prob" value="0.01" />

    <param name="ros/skip_exceeding_lsr_vals" value="false" />
    <param name="tinySLAM/hole_width" value="0.5" />

    <!-- Map setup-->
    <param name="slam/map/height_in_meters" value="100" />
    <param name="slam/map/width_in_meters" value="100" />
    <param name="slam/map/meters_per_cell" value="0.1" />

    <!-- Scan Matching -->
    <param name="~slam/scmtch/oope/type" value="obstacle" />

    <!-- Scan Matcher setup -->
    <param name="slam/scmtch/type" value="MC" />
    <param name="slam/scmtch/MC/dispersion/translation" value="0.2" />
    <param name="slam/scmtch/MC/dispersion/rotation" value="0.1" />
    <param name="slam/scmtch/MC/dispersion/failed_attempts_limit" value="20" />
    <param name="slam/scmtch/MC/attempts_limit" value="100" />
    <!-- <param name="slam/scmtch/MC/seed" value="666666" /> -->

    <!-- Internal constants for ros-->
    <param name="ros/rviz/map_publishing_rate" value="5.0" />
    <param name="ros/tf/buffer_duration" value="5.0" /> <!-- in seconds-->
    <param name="ros/filter_queue_size" value="1000" />
    <param name="ros/subscribers_queue_size" value="1000" />
  </node>

  <node name="player" pkg="rosbag" type="play"
        args="--delay=3 --clock --rate $(arg rate) $(arg path)" />

  <node name="rviz" pkg="rviz" type="rviz" respawn="true"
        args="-d $(find slam_fmwk)/rviz/debug.rviz" />

</launch>
